imnodes.cpp

```diff
@@ -976,6 +997,34 @@ void ClickInteractionUpdate(ImNodesEditorContext& editor)
         }
     }
     break;
+    case ImNodesClickInteractionType_MiniMapPanning:
+    {
+        const bool dragging = GImNodes->AltMouseDragging;
+
+        if (dragging)
+        {
+            editor.Panning += ImGui::GetIO().MouseDelta / GImNodes->MiniMapZoom;
+        }
+        else
+        {
+            editor.ClickInteraction.Type = ImNodesClickInteractionType_None;
+        }
+    }
+    break;
+    case ImNodesClickInteractionType_MiniMapZooming:
+    {
+        GImNodes->MiniMapZoom = fmax(0.05f, fmin(GImNodes->MiniMapZoom + 0.1f * GImNodes->MiniMapZoom * GImNodes->AltMouseScrollDelta, 1.f));
+        editor.ClickInteraction.Type = ImNodesClickInteractionType_None;
+    }
+    break;
+    case ImNodesClickInteractionType_MiniMapSnapping:
+    {
+        editor.Panning += GImNodes->MiniMapRectSnappingOffset;
+        GImNodes->MiniMapRectSnappingOffset = ImVec2(0.f, 0.f);
+
+        editor.ClickInteraction.Type = ImNodesClickInteractionType_None;
```

Comment:
I think it might feel a bit more consistent if the actual snap-to-node happens when the mouse button is released. In other click-based interactions, such as clicking checkboxes, buttons, and nodes, the action only happens on mouse release.

Reply:
Will change to snap on mouse release.